test/functional/sqlsrv/sqlsrv_ae_output_param_sqltype_numeric.phpt

```diff
@@ -0,0 +1,225 @@
+    $r;
+    // convert input values to strings for decimals and numerics
+    if ($dataTypes == "decimal(18,5)" || $dataTypes == "numeric(10,5)") {
+        $stmt = AE\insertRow($conn, $tbname, array( $colMetaArr[0]->colName => (string) $inputValues[0], $colMetaArr[1]->colName => (string) $inputValues[1] ), $r);
+    } else {
+        $stmt = AE\insertRow($conn, $tbname, array( $colMetaArr[0]->colName => $inputValues[0], $colMetaArr[1]->colName => $inputValues[1] ), $r);
+    }
+    if ($r === false) {
+        is_incompatible_types_error($dataType, "default type");
+    }
+
+    foreach($directions as $direction) {
+        echo "Testing as $direction:\n";
+
+        // test each SQLSRV_SQLTYPE_ constants
+        foreach ($sqlTypes as $sqlType) {
+
+            if (!AE\isColEncrypted()) {
+                $isCompatible = false;
+                foreach ($compatList[$dataType] as $compatType) {
+                    if (stripos($compatType, $sqlType) !== false) {
+                        $isCompatible = true;
+                    }
+                }
+                // 22018 is the SQLSTATE for any incompatible conversion errors
+                if ($isCompatible && sqlsrv_errors()[0]['SQLSTATE'] == 22018) {
+                    echo "$sqlType should be compatible with $dataType\n";
+                    $success = false;
+                }
+            } else {
+                // skip unsupported datetime types
+                if (!isDateTimeType($sqlType)) {
+                    $sqlTypeConstant = get_sqlType_constant($sqlType);
+
+                    // Call store procedure
+                    $outSql = AE\getCallProcSqlPlaceholders($spname, 2);
+                    if ($sqlType == 'SQLSRV_SQLTYPE_FLOAT' || $sqlType == 'SQLSRV_SQLTYPE_REAL') {
+                        $c_detOut = 0.0;
+                        $c_randOut = 0.0;
+                    } else {
+                        $c_detOut = 0;
+                        $c_randOut = 0;
+                    }
+                    $stmt = sqlsrv_prepare($conn, $outSql, 
+                        array(array( &$c_detOut, constant($direction), null, $sqlTypeConstant),
+                        array(&$c_randOut, constant($direction), null, $sqlTypeConstant)));
+
+                    if (!$stmt) {
+                        die(print_r(sqlsrv_errors(), true));
+                    }							
+                    sqlsrv_execute($stmt);
+                    $errors = sqlsrv_errors();
+                    
+                    if (!empty($errors)) {
+                        if (stripos("SQLSRV_SQLTYPE_" . $dataType, $sqlType) !== false) {
+                            var_dump(sqlsrv_errors());
+                            $success = false;               
+                        }	
+                    }
+                    else {
+                        print("c_det: " . $c_detOut . "\n");
+                        print("c_rand: " . $c_randOut . "\n");

```

Comment:
If you're already comparing the output with the inputValues (line 110 and 115), I don't think printing the values out here is necessary. Especially for the float values, sometimes in some platform some flavor the float returned may be slightly different from the expected output. Comparing the difference with epsilon is good enough.